### ISSUE TITLE
Package fails for Julia 1.6

### ISSUE THREAD

**Issue by adknudson** [adknudson/NearestCorrelationMatrix.jl#4]
Currently the package fails for Julia 1.6.

```julia
ERROR: LoadError: DimensionMismatch("cannot broadcast array to have fewer dimensions")
Stacktrace:
  [1] check_broadcast_shape(#unused#::Tuple{}, Ashp::Tuple{Base.OneTo{Int64}})
    @ Base.Broadcast .\broadcast.jl:518
  [2] check_broadcast_shape(shp::Tuple{Base.OneTo{Int64}}, Ashp::Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}})
    @ Base.Broadcast .\broadcast.jl:521
  [3] check_broadcast_axes
    @ .\broadcast.jl:523 [inlined]
  [4] instantiate
    @ .\broadcast.jl:269 [inlined]
  [5] materialize!
    @ .\broadcast.jl:894 [inlined]
  [6] materialize!
    @ .\broadcast.jl:891 [inlined]
  [7] _precondition_matrix!(c::Vector{Float32}, Ω₀::Matrix{Float32}, P::Matrix{Float32})
```

**Comment by peteristhegreat** [comment]
Here's the fix to use `vec` in `newton.jl` to satisfy the Julia 1.6 compiler.  This resolves the error (at least for precompiling and going through its runtests).

```
@@ -267,7 +267,7 @@ function _precondition_matrix!(
         c .= transpose(sum(H₁, dims=1)).^2 + 2 * sum(H12 .* transpose(H₂), dims=2)
     else
         H12 = (1.0 .- Ω₀) * H₂
-        c .= transpose(sum(H, dims=1)).^2 - transpose(sum(H₂, dims=1)).^2 - 2 * transpose(sum(H₁ .* H12, dims=1))
+        c .= vec(sum(H, dims=1)).^2 - vec(sum(H₂, dims=1)).^2 - 2 * vec(sum(H₁ .* H12, dims=1))
     end

     ϵ = T(1e-8)
@@ -346,7 +346,7 @@ function _jacobian!(
         HT₁ = P₂ * transpose(Ω)
         HT₂ = P₂ * transpose(H₂) * P₂ + P₁ * Ω

-        w .= x .* (1 + perturbation) - sum(P .* [HT₁ HT₂], dims=2)
+        w .= x .* (1 + perturbation) - vec(sum(P .* [HT₁ HT₂], dims=2))
         return w
     end
 end
 ```

**Comment by adknudson** [comment]
Thank you for the suggested fix! I will reopen this issue and try out your suggestion as it would be nice to support the LTS version of Julia.

**Comment by peteristhegreat** [comment]
BTW, I patched the 0.2.2 version before all the new deps were added for it
to work in my 1.6 environment. I haven't tried to get the new deps added in
March 2024 to all work and play nice with 1.6.

**Comment by adknudson** [comment]
@peteristhegreat Now that 1.11 is released and 1.10 is the new LTS version, do you still plan to have a 1.6 environment?

**Comment by peteristhegreat** [comment]
Eventually our larger project will switch up to 1.10.

I have the patched version working and just have it with Pkg.develop in my
env. No worries.  I mostly posted my solution in case there was someone
else still in an older version of Julia and wanted to get it to work.
